File: Programming/ecobike_rental/lib/data/model/card.dart
import '../db/database.dart';
import 'models.dart';

class Card implements Model {
  Card({
    this.id,
    this.cardCode,
    this.owner,
    this.cvvCode,
    this.dateExpired,
    this.paymentMethod,
  });

  Card.empty();

  Card.fromMap(Map<String, dynamic> map) {
    id = map[Card.key];
    renderId = map['renter_id'];
    paymentMethod = map['payment_method'];
    cvvCode = map['cvv'];
    dateExpired = map['expiration_date'];
  }

  static String tableName = "Card";
  static String key = "card_id";
  int id;
  String paymentMethod;
  String cardCode;
  String owner;
  int renderId;
  Renter renter;
  int cvvCode;
  String dateExpired;

  @override
  Map<String, dynamic> toMap() {
    return {
      Card.key: id,
      'renter_id': renderId,
      'payment_method': paymentMethod,
      'cvv': cvvCode,
      'expiration_date': dateExpired
    };
  }

  @override
  String getTableName() {
    return Card.tableName;
  }
}

File: Programming/ecobike_rental_server/TKXDPM_API/Model/Card.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using AutoMapper;

namespace TKXDPM_API.Model
{
    public class Card
    {
        [Key] public int CardId { get; set; }
        [Column(TypeName = "varchar(255)")] public string CardCode { get; set; }
        public int RenterId { get; set; }
        public Renter Renter { get; set; }
        [Column(TypeName = "varchar(255)")] public string PaymentMethod { get; set; }
        public int Cvv { get; set; }
        public DateTime ExpirationDate { get; set; }
    }

    public class CardRequest
    {
        public int RenterId { get; set; }
        public string CardCode { get; set; }
        public string PaymentMethod { get; set; }
        public int Cvv { get; set; }
        public DateTime ExpirationDate { get; set; }
    }

    public class CardResponse
    {
        public static CardResponse CreateByRenter(Renter renter, IMapper mapper)
        {
            var cardResponse = mapper.Map<CardResponse>(renter.Card);
            cardResponse.Renter = mapper.Map<RenterResponse>(renter);
            return cardResponse;
        }

        public int CardId { get; set; }
        public RenterResponse Renter { get; set; }
        public string PaymentMethod { get; set; }
        public int Cvv { get; set; }
        public string ExpirationDate { get; set; }
    }
}